view/utils.js

```diff
@@ -99,10 +99,10 @@ function removeSpinner(spinnerID) {
  * @param {String} url The url to get from
  * @returns {Promise} For use in chaining.
  */
-function getPromiseWithUrl(url) {
+function getPromiseWithUrl(url, headers) {
```

Comment:
Can you please add a function-level description for this function and note that when you do not pass in any headers, caching is off, but that you can send in your own url-specific headers if you want to override that and turn caching on